### DISCUSSION TITLE
Is it possible to change the directory where Spot stores and executes scripts on a remote host?

### DISCUSSION THREAD

**Discussion by vo-va** [umputun/spot#232] [Q&A] [answered]
Some operating systems might mark certain paths or file systems as non-executable (for example, this article explains this for Container-Optimized OS https://cloud.google.com/container-optimized-os/docs/concepts/disks-and-filesystem). Because of that (/tmp is non-executable on Container-Optimized OS), and since Spot saves scripts by default to the /tmp directory and tries to execute them from there, it always fails when the script is created using the YAML multiline approach.

I guess it is possible to mitigate this by copying the script from the local computer to the remote host in the desired place and then executing it. 

```
2024/10/23 16:03:38.176 [DEBUG] file mode for /var/folders/tv/l0l32mdx14zgwrm2ssn1hmwh0000gp/T/spot-script4169099696: 0700
2024/10/23 16:03:38.475 [INFO]  uploaded /var/folders/tv/l0l32mdx14zgwrm2ssn1hmwh0000gp/T/spot-script4169099696 to 127.0.0.1:/tmp/.spot-360257565625078720/spot-script4169099696 in 393.551334ms
2024/10/23 16:03:38.475 [DEBUG] run /bin/sh -c /tmp/.spot-360257565625078720/spot-script4169099696
2024/10/23 16:03:38.475 [DEBUG] run ssh command "/bin/sh -c /tmp/.spot-360257565625078720/spot-script4169099696" on 127.0.0.1:2222
2024/10/23 16:03:38.505 [DEBUG] > /bin/sh -c /tmp/.spot-360257565625078720/spot-script4169099696
2024/10/23 16:03:38.538 [WARN]  ! /bin/sh: line 1: /tmp/.spot-360257565625078720/spot-script4169099696: Permission denied
2024/10/23 16:03:38.916 [INFO]  deleted recursively /tmp/.spot-360257565625078720
```

**Comment by umputun** [comment, marked as answer]
see https://github.com/umputun/spot/pull/253

**Reply by vo-va** [reply]
Thank you!
